### PR TITLE
drivers/lis3mdl: adopt new parameters scheme and provide SAUL adaption

### PR DESCRIPTION
**Pull request by aabadie**
While working on #7937, I noticed that some drivers implementation where very incomplete regarding new initialization parameter scheme and SAUL adaption.

This PR provides an update of the init API and SAUL adaption for LIS3MDL 3-axis magnetometer device.

It also enables it for the limifrog-v1 board when the saul module is loaded.

Also related to #7519

### CODE REVIEW

**Comment by aabadie** [comment]
@PeterKietzmann @smlng unfortunately I don't have the hardware for testing this one. Maybe you can help ?

**Comment by aabadie** [comment]
@PeterKietzmann, the board adapted by #7584 has one. I'll try with this one.

File: drivers/lis3mdl/include/lis3mdl_params.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+{
+    LIS3MDL_PARAMS
+};
+
+/**
+ * @brief   Additional meta information to keep in the SAUL registry
+ */
+static const saul_reg_info_t lis3mdl_saul_info[] =
+{
+    { .name = "lis3mdl" }
```
Comment (smlng) [inline drivers/lis3mdl/include/lis3mdl_params.h]: shouldn't this be a macro like `LIS3MDL_SAUL_INFO`, as you suggested yourself to me in #7280 😉

File: drivers/lis3mdl/lis3mdl.c [hunk 2]
```diff
@@ -32,6 +32,9 @@
 
 #define GAUSS_DIVIDER      (1000)
 
+#define DEV_I2C      (dev->params.i2c)
```
Comment (smlng) [inline drivers/lis3mdl/lis3mdl.c]: match indention with above `GAUSS_DIVIDER`

File: drivers/lis3mdl/include/lis3mdl_params.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2017 Inria
```
Comment (smlng) [inline drivers/lis3mdl/include/lis3mdl_params.h]: missing newline below copyright

File: drivers/lis3mdl/lis3mdl.c [hunk 4]
```diff
@@ -120,14 +116,14 @@ void lis3mdl_read_mag(const lis3mdl_t *dev, lis3mdl_3d_data_t *data)
...
+    i2c_release(DEV_I2C);
 }
 
 void lis3mdl_read_temp(const lis3mdl_t *dev, int16_t *value)
 {
-    i2c_acquire(dev->i2c);
-    i2c_read_regs(dev->i2c, dev->addr, LIS3MDL_TEMP_OUT_L_REG, (uint8_t*)value, 2);
-    i2c_release(dev->i2c);
+    i2c_acquire(DEV_I2C);
+    i2c_read_regs(DEV_I2C, DEV_ADDR, LIS3MDL_TEMP_OUT_L_REG, (uint8_t*)value, 2);
```
Comment (smlng) [inline drivers/lis3mdl/lis3mdl.c]: is it safe to read 2x 1-byte into a 2-bytes datatype, anyway?

Comment (smlng) [reply drivers/lis3mdl/lis3mdl.c]: not memory but value-wise?

File: sys/auto_init/saul/auto_init_lis3mdl.c [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+/**
+ * @brief   Allocate memory for the device descriptors
+ */
+static lis3mdl_t lis3mdl_devs[LIS3MDL_NUM];
+
+/**
+ * @brief   Memory for the SAUL registry entries
+ */
+static saul_reg_t saul_entries[LIS3MDL_NUM * 2];
```
Comment (smlng) [inline sys/auto_init/saul/auto_init_lis3mdl.c]: why 2?

File: drivers/lis3mdl/lis3mdl.c (hunk 4 above)
Comment (aabadie) [reply drivers/lis3mdl/lis3mdl.c]: I admit, I didn't look at that in detail but it looks valid: *value parameter of `read_temp` function is the address of the 2-bytes datatype variable and (uint8_t*)value cast it to an address of 2-bytes array.

**Comment by aabadie** [comment]
@smlng, comments addressed

**Comment by aabadie** [comment]
@smlng @PeterKietzmann, I could test this PR with success (after some changes in the default parameters) with a [ST MEMS shield](http://www.st.com/en/ecosystems/x-nucleo-iks01a1.html). It works with both `tests/driver_lis3mdl` and `examples/default` (saul).

**Pull request merged**